rust/extractor/src/main.rs

```diff
@@ -42,6 +42,9 @@ fn main() -> anyhow::Result<()> {
 
     let config = CargoConfig {
         sysroot: Some(RustLibSource::Discover),
+        target_dir: std::env::var("CODEQL_EXTRACTOR_RUST_SCRATCH_DIR")
+            .map(|p| ra_ap_paths::Utf8PathBuf::from(p).join("target"))
+            .ok(),
```

Comment:
I would rather add it to `config.rs`, like the other directories, so that we can also pass `--scratch_dir` to tweak this.

Also, my memory is hazy, but I think that despite the extractor-specific env variable name, the scratch dir is actually shared when building in cluster mode, so we might need to use `rust/target` for robustness, but I'll try to double check.